File: 077.LeetCode@129_Sum_Root_to_Leaf_Numbers/077.LeetCode@129_Sum_Root_to_Leaf_Numbers.java
class Solution {
    public int sumNumbers(TreeNode root) {
        if (root == null) return 0;
        String a = "";
        return find(root, a);
    }

    private int find(TreeNode root, String a) {
        if (root == null) return 0;
        if (root.left == null && root.right == null) return Integer.parseInt(a + root.val);
        return find(root.left, a + root.val) + find(root.right, a + root.val);
    }
}

File: 077.LeetCode@129_Sum_Root_to_Leaf_Numbers/077.LeetCode@129_Sum_Root_to_Leaf_Numbers.py
class Solution:
    def sumNumbers(self, root):
        if not root:
            return 0
        return self.find(root, "")

    def find(self, root, a):
        if not root:
            return 0
        if not root.left and not root.right:
            return int(a + str(root.val))
        return self.find(root.left, a + str(root.val)) + self.find(root.right, a + str(root.val))